doc/static_route/SONiC_static_route_hdl.md

```diff
@@ -0,0 +1,546 @@
+8) "Vrf-BLUE,Vrf-RED"
+9) "blackhole@"
+10) "false,false"
+```
+FRR running config:
+```
+!
+vrf Vrf-test
+ ip route 1.1.0.0/16 2.2.2.2 Ethernet0 10 nexthop-vrf Vrf-BLUE
+ ip route 1.1.0.0/16 Ethernet4 20 nexthop-vrf Vrf-RED
+ exit-vrf
+!
+```
+### 3.7.2 Configuration Reload
+All static route configurations were persistently stored in config DB STATIC_ROUTE table. After BGP container restarts, the configuration in DB needs to be re-applied. This configuration reload is done by generating staticd.conf file before FRR staticd daemon starts. A jinja template file will be used to map table entries to fill in staticd.conf file. The generated conf file is loaded by FRR daemon to configure static routes to system.
+
+# 4 Unit Test and automation
+The following test cases will be tested using CLI/REST/gNMI management interfaces and spytest will be incremeted.
+
+|Test Cases (done on default instance and data VRF)| Test Result |
+| :------ | :----- |
+|create route for a prefix with NH IP addresses|    verify the route is created|
+|create route for a prefix with NH IP address and interface| verify route is created|
+|create route for a prefix with interface only| verify route is created|
+|create route for a prefix with NH IP addresses and NH VRF|    verify the route is created|
+|create route for a prefix with NH IP address, interface and NH VRF| verify route is created|
+|create route for a prefix with interface and NH VRF| verify route is created|
+|create route for a prefix with NH IP addresses and distance|    verify the route is created|
+|create route for a prefix with NH IP address, interface and distance | verify route is created|
+|create route for a prefix with interface and distance | verify route is created|
+|create route for a prefix with NH IP addresses, NH VRF and distane |    verify the route is created|
+|create route for a prefix with NH IP address, interface and distance| verify route is created|
+|create route for a prefix with interface, NH VRF and distance | verify route is created|
+|create route for a prefix with blackhole and distance |verify route is created
+|CLI: show ip route static|   verify static routes are displayed with flag "S" (i.e Static)|
+|Delete route for a prefix with NH IP addresses| verify the route is deleted|
+|Delete route for a prefix with NH IP address and interface| verify route is deleted|
+|Delete route for a prefix with interface only| verify route is deleted|
+|Delete route for a prefix with NH IP addresses and NH VRF|    verify the route is deleted|
+|Delete route for a prefix with NH IP address, interface and NH VRF| verify route is deleted|
+|Delete route for a prefix with interface and NH VRF| verify route is deleted|
+|Delete route for a prefix with blackhole| verify route is deleted|
+
+| Test cases for management static routes |Test Result |
+| :------ | :----- |
+|create route in default instance  for a prefix with NH IP address and interface as Management0 | verify route is created|

```

Comment:
This route should be pushed to HW ?

Reply:
orchagent doesn't push any route with eth0 to HW so this will not go to HW.

Reply:
If a data traffic hits in default VRF but needs to go to the Mgmt VRF ==> how the forwarding should work here ? Please explain in general how data forwarding will work from non-mgmt VRF to mgmt VRF